lib/modules/manager/devbox/artifacts.ts

```diff
@@ -1,42 +1,58 @@
 import is from '@sindresorhus/is';
+import semver from 'semver';
 import { logger } from '../../../logger';
 import { exec } from '../../../util/exec';
 import type { ExecOptions } from '../../../util/exec/types';
 import { getSiblingFileName, readLocalFile } from '../../../util/fs';
 import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
 
-export async function updateArtifacts(
-  updateConfig: UpdateArtifact,
-): Promise<UpdateArtifactsResult[] | null> {
-  const lockFileName = getSiblingFileName(
-    updateConfig.packageFileName,
-    'devbox.lock',
-  );
+export async function updateArtifacts({
+  config: { constraints, env, isLockFileMaintenance, updateType },
+  packageFileName,
+  updatedDeps,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  const lockFileName = getSiblingFileName(packageFileName, 'devbox.lock');
   const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
   if (!existingLockFileContent) {
     logger.debug('No devbox.lock found');
     return null;
   }
+
+  const supportsNoInstall = constraints?.devbox
+    ? semver.satisfies(constraints.devbox, '>=0.14.0')
```

Comment:
```suggestion
    ? semver.intersects(constraints.devbox, '>=0.14.0')
```
maybe this is safer if constraint is a range. 🤔